dev-cpp/sol2/sol2-3.3.0.ebuild

```diff
@@ -0,0 +1,73 @@
+# Copyright 2021-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+LUA_COMPAT=( lua5-{1,3,4} luajit )
+inherit cmake lua
+
+DESCRIPTION="Header-only C++ <-> Lua API wrapper"
+HOMEPAGE="https://github.com/ThePhD/sol2"
+SRC_URI="https://github.com/ThePhD/sol2/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
+REQUIRED_USE="test? ( ${LUA_REQUIRED_USE} )"
```

Comment:
Oh, just noticed this.

This lets you do `LUA_TARGETS="" FEATURES="-test" emerge -1 sol2` and then fail miserably. These lua_foreach_impl aren't guarded behind tests anymore in this ebuild.

Reply:
Guess previous ebuild was also missing a pkg_setup() { use test ... } guard too.

Reply:
```
 * No Lua implementation selected for the build. Please add one
 * of the following values to your LUA_TARGETS
 * (in make.conf or package.use):
 * 
 * lua5-1 lua5-3 lua5-4 luajit
```

Looks correct to me

Reply:
> Looks correct to me

Not really, REQUIRED_USE is supposed to serve its purpose rather than have the eclass die from unexpected use. Also confusing for scripts, tinderbox can end up filling bugs about these given they're not being told what to set the normal way (not that it happens if the default impl lua5.1 is an option, given they always have something that works set).

Either way it's fine now that the `test?` is gone and LUA_DEPS are unconditional.

wrt src_install comment, unfortunate a headers-only package need to find lua even without tests now and loop through all targets when I don't think it does anything useful without tests, but given it'll ultimately need lua either way and most people only set 1 target I guess it's not so bad :man_shrugging: